packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -393,6 +393,8 @@ class AppDomain extends Domain {
 
     if (options.debuggingEnabled) {
       connectionInfoCompleter = new Completer<DebugConnectionInfo>();
+      // Callback hookups' Futures are uninteresting.
+      // ignore: unawaited_futures
       connectionInfoCompleter.future.then<Null>((DebugConnectionInfo info) {
```

Comment:
It may be interesting to introduce a `Future` like class, but one that doesn't complete with an error. This would be the equivalent of a method that isn't spec'd to exit with an exception - the class could enforce this by API.